### PR TITLE
Automated Flannel version update 0.21.2

### PR DESCRIPTION
**Pull request by replicated-ci-kurl**
Automated changes by the [cron-flannel-update](https://github.com/replicatedhq/kURL/blob/main/.github/workflows/update-flannel.yaml) GitHub action

```release-note
Adds [Flannel add-on](https://kurl.sh/docs/add-ons/flannel) version 0.21.2.
```

### CODE REVIEW

File: addons/flannel/0.21.2/Manifest [hunk 1]
```diff
@@ -0,0 +1,2 @@
+image flannel-flannel docker.io/flannel/flannel:v0.20.2
```
Comment (emosbaugh) [inline addons/flannel/0.21.2/Manifest:1]: This is still using the wrong version.

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/flannel/0.21.2/Manifest (hunk 1 above)
Comment (camilamacedo86) [inline addons/flannel/0.21.2/Manifest:1]:
```suggestion
image flannel-flannel docker.io/flannel/flannel:v0.21.2
```

Comment (camilamacedo86) [inline addons/flannel/0.21.2/Manifest:1]:
```suggestion
image flannel-flannel docker.io/flannel/flannel:v0.21.2
```

**Review by camilamacedo86 — APPROVED** [review]

File: addons/flannel/0.21.2/yaml/kube-flannel.yml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        - cp
+        args:
+        - -f
+        - /flannel
+        - /opt/cni/bin/flannel
+        volumeMounts:
+        - name: cni-plugin
+          mountPath: /opt/cni/bin
+      - name: install-cni
+        image: docker.io/flannel/flannel:v0.20.2
```
Comment (emosbaugh) [inline addons/flannel/0.21.2/yaml/kube-flannel.yml:145]: This is not the correct image.

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/flannel/0.21.2/yaml/kube-flannel.yml (hunk 2 above)
Comment (camilamacedo86) [inline addons/flannel/0.21.2/yaml/kube-flannel.yml:145]:
```suggestion
        image: docker.io/flannel/flannel:v0.21.2
```

File: addons/flannel/0.21.2/yaml/kube-flannel.yml [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+       #image: docker.io/rancher/mirrored-flannelcni-flannel:v0.20.2
```
Comment (camilamacedo86) [inline addons/flannel/0.21.2/yaml/kube-flannel.yml:146]:
```suggestion
       #image: docker.io/rancher/mirrored-flannelcni-flannel:v0.21.2
```

File: addons/flannel/0.21.2/yaml/kube-flannel.yml [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        - /etc/kube-flannel/cni-conf.json
+        - /etc/cni/net.d/10-flannel.conflist
+        volumeMounts:
+        - name: cni
+          mountPath: /etc/cni/net.d
+        - name: flannel-cfg
+          mountPath: /etc/kube-flannel/
+      containers:
+      - name: kube-flannel
+        image: docker.io/flannel/flannel:v0.20.2
```
Comment (camilamacedo86) [inline addons/flannel/0.21.2/yaml/kube-flannel.yml:160]:
```suggestion
        image: docker.io/flannel/flannel:v0.21.2
```

File: addons/flannel/0.21.2/yaml/kube-flannel.yml [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+       #image: docker.io/rancher/mirrored-flannelcni-flannel:v0.20.2
```
Comment (camilamacedo86) [inline addons/flannel/0.21.2/yaml/kube-flannel.yml:161]:
```suggestion
       #image: docker.io/rancher/mirrored-flannelcni-flannel:v0.21.2
```

Comment (camilamacedo86) [inline addons/flannel/0.21.2/yaml/kube-flannel.yml:161]: This one needs to be 0.20.2, see : https://hub.docker.com/r/rancher/mirrored-flannelcni-flannel/tags

File: addons/flannel/0.21.2/yaml/kube-flannel.yml (hunk 2 above)
Comment (camilamacedo86) [reply addons/flannel/0.21.2/yaml/kube-flannel.yml]: Good catcher. I tried to check all manifests and updated then.
Please, let me know if you find any other place that is outdated.

File: addons/flannel/0.21.2/Manifest (hunk 1 above)
Comment (camilamacedo86) [reply addons/flannel/0.21.2/Manifest:1]: The script re-run and tghe values were updated again we will need to create a PR based on this one and probably see how to fix the script as well.

File: addons/flannel/0.21.2/yaml/kube-flannel.yml (hunk 5 above)
Comment (chris-sanders) [reply addons/flannel/0.21.2/yaml/kube-flannel.yml:161]: Can you explain this that link doesn't make it obvious to me why this needs to be 20.2.

**Pull request closed without merging**